compiler/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala

```diff
@@ -1366,9 +1364,9 @@ class TreeUnpickler(reader: TastyReader,
       val path = sourcePathAt(addr)
       if (path.nonEmpty) {
         val sourceFile = ctx.getSource(path)
-        lineSizesUnpicklerOpt match
-          case Some(lineSizesUnpickler) =>
-            sourceFile.setLineIndices(lineSizesUnpickler.lineIndices)
+        posUnpicklerOpt match
+          case Some(posUnpickler) =>
+            sourceFile.setLineIndicesFromLineSizes(posUnpickler.lineSizes)
```

Comment:
There is a potential issue here: the relative path of a file is not a unique identifier of the file, there might be path conflicts in the Scala ecosystem.

If we also store the hash of source files, that would be better.

Reply:
How would you use that hash?

Reply:
Maybe `ctx.getSource(path)` can take hash as an argument? I haven't thought through the details.

Reply:
The hash would be on the contents of the source? Then we would need to read all the sources eagerly when we add them to the context. That may be quite expensive.

Reply:
When we compile files, we already have the file contents in memory.

Reply:
Actually, it is only loaded when we first access something the depends on the source https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/dotc/util/SourceFile.scala#L43.

Reply:
When we compile Scala files, the contents for the Scala files will be forced.